packages/rest-client/src/services/store/store.ts

```diff
@@ -99,13 +109,65 @@ export function keyResourcesById(data: Resource[]): {
 }
 
 /**
- * Converts a resource to a record.
+ * Takes a list of Resources containing both top-level records and related records and
+ * flattens them into a list of Records. Flattens the attributes and relationships keys.
+ * Merges the attribute data of the related records into the top-level records.
  */
-export function convertResourceToRecord(resource: Resource): Record {
-  return {
-    id: resource.id,
-    __schema: resource.__schema,
-    ...resource.attributes,
-    // @todo relationships
+export function flattenResourcesIntoRecords(
+  resources: Resource[],
+  topLevelSchemaName: string,
+): Record[]
+export function flattenResourcesIntoRecords(
+  resources: Resource[],
+  topLevelRecordSchemaName: string,
+  id: string,
+): Record | undefined
+export function flattenResourcesIntoRecords(
```

Comment:
👍  this is pretty cool

Reply:
took me a bit to figure out the proper syntax 🤣